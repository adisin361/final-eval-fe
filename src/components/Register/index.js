import React from 'react';
import bgImg from '../../assets/undraw-upload-re-pasx_2023-03-09/undraw-upload-re-pasx.png';
import './style.css';
import { useNavigate } from 'react-router-dom';
// import { makeAuthRequest } from '../../utils/makeRequest';
// import { REGISTER_URL } from '../../constants/apiEndpoints';
const Register = () => {
  const navigate = useNavigate();

  return (
    <>
      <div className='login-page'>
        <div className='image-section'>
          <p>Design APIs fast,</p>
          <p>Manage content easily</p>
          <img src={bgImg} />
        </div>

        <div className='login-section'>
          <p>Register your CMS+ account now!</p>
          <div className='login-form'>
            <p>Email</p>
            <input type='text' />

            <p>Password</p>
            <input type='password' />
            <br />
            <button>Register</button>
            <p id='forgot-pwd' className='have-account' onClick={() => {
              navigate('/login');
            }}>Already have an account</p>
          </div>
        </div>
      </div>
    </>
  );
};

export default Register;