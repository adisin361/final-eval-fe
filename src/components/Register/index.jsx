/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import bgImg from '../../assets/undraw-upload-re-pasx_2023-03-09/undraw-upload-re-pasx.png';
import './style.css';
import { useNavigate } from 'react-router-dom';
import { makeAuthRequest } from '../../utils/makeRequest';
import { REGISTER_URL } from '../../constants/apiEndPoints';
const Register = () => {
  const navigate = useNavigate();
  const [user, setUser] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState(null);

  const handleSubmit = async () => {
    console.log(user);
    console.log(password);
    try {
      await makeAuthRequest(REGISTER_URL, {
        data: {
          username: user,
          password: password,
        },
      });
      navigate('/login');
    } catch (error) {
      setError(error);
    }
  };

  return (
    <>
      <div className='login-page'>
        <div className='image-section'>
          <p>Design APIs fast,</p>
          <p>Manage content easily</p>
          <img src={bgImg} />
        </div>

        <div className='login-section'>
          <p>Register your CMS+ account now!</p>
          <div className='login-form'>
            <p>Email</p>
            <input type="email"
              name="username"
              value={user}
              onChange={(event) => setUser(event.target.value)} />

            <p>Password</p>
            <input type="password"
              name="password"
              value={password}
              onChange={(event) => setPassword(event.target.value)} />
            <br />
            <button type="submit"
              disabled={user === '' || password === ''}
              onClick={handleSubmit}>Register</button>
            <p id='forgot-pwd' className='have-account' onClick={() => {
              navigate('/login');
            }}>Already have an account</p>
          </div>
        </div>
      </div>
    </>
  );
};

export default Register;