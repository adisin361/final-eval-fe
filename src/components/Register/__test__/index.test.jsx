import React from 'react';
import Register from '..';
import { render } from '@testing-library/react';

const mockedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockedNavigate,
}));

describe('Login Page', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Register />);
    expect(asFragment()).toMatchSnapshot();
  });
});