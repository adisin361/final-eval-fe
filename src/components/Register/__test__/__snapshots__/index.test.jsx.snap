// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="login-page"
  >
    <div
      class="image-section"
    >
      <p>
        Design APIs fast,
      </p>
      <p>
        Manage content easily
      </p>
      <img
        src="undraw-upload-re-pasx.png"
      />
    </div>
    <div
      class="login-section"
    >
      <p>
        Register your CMS+ account now!
      </p>
      <div
        class="login-form"
      >
        <p>
          Email
        </p>
        <input
          name="username"
          type="email"
          value=""
        />
        <p>
          Password
        </p>
        <input
          name="password"
          type="password"
          value=""
        />
        <br />
        <button
          disabled=""
          type="submit"
        >
          Register
        </button>
        <p
          class="have-account"
          id="forgot-pwd"
        >
          Already have an account
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
